docs/devices.rst

```diff
@@ -0,0 +1,147 @@
+    >>> for d in query:
+    ...     print(f"{d.name} - {d.last_internal_ip_address}")
+    DESKTOP-A19 - 10.0.2.44
+    DESKTOP-Q210 - 10.10.25.169
+    DESKTOP-Q211 - 10.10.25.170
+    DESKTOP-Q211B - 10.10.25.180
+    EVALUATION-1 - 10.0.2.51
+    EVALUATION-2 - 10.0.2.52
+    STAGING-1A - 192.168.1.99
+    ZZIGNORE-1 - 10.0.3.74
+
+The criteria supported in the ``where()`` and ``add_criteria()`` query methods are too numerous to enumerate here;
+please see
+`the Developer Network documentation <https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/devices-api/#search-devices>`_
+for more details.
+
+The results of a search query can also be exported::
+
+    >>> from cbc_sdk import CBCloudAPI
+    >>> api = CBCloudAPI(profile='sample')
+    >>> from cbc_sdk.platform import Device
+    >>> query = api.select(Device).where("os:WINDOWS")
+    >>> query.add_criteria('target_priority', ['LOW']).add_criteria('virtualization_provider', ['VirtualBox'])
+    >>> job = query.export()
+    >>> csv_report = job.get_output_as_string()
+    >>> # can also get the output as a file or as enumerated lines of text
+
+Search Scrolling
+++++++++++++++++
+
+A Device Search request can return no more than 10,000 items at a time.  Some customers may have more endpoints than
+that; to return *all* devices, you can use the ``scroll()`` method on the query to continue searching after all devices
+that have been previously returned.  This snippet illustrates the technique::
+
+    # assume "api" is your CBCloudAPI reference
+    query = api.select(Device)
+    # add search terms and/or criteria to the query (not shown here)
+    for d in query:
+        do_something_with_device(d)  # whatever you need for each device
+    while query.num_remaining > 0:
+        query.scroll()  # default is 10,000 devices at a time
+        for d in query:
+            do_something_with_device(d)
+
+Device Actions
+--------------
+
+Most device actions in the Carbon Black Cloud can be performed on a single device through the ``Device`` object,
+on multiple devices specified by ID, or on the results of a device query.
+
+Bypass Enable/Disable
++++++++++++++++++++++
+
+Setting a device to *bypass* disables all enforcement on the device; its sensor stops sending data to the Carbon Black
+Cloud.
+
+Setting bypass on a single device::
+
+    >>> # assume "api" is your CBCloudAPI reference
+    >>> d = api.select(Device, 12345)
+    >>> d.bypass(True)
+
+Setting bypass on multiple devices::
+
+    >>> # assume "api" is your CBCloudAPI reference
+    api.device_bypass([1001, 1002, 1003], True)
+
+Setting bypass on the results of a device search::
+
+    >>> # assume "api" is your CBCloudAPI reference
+    query = api.select(Device)
+    # add search terms and/or criteria to the query (not shown here)
+    query.bypass(True)
+
+Quarantine
+++++++++++
+
+A device that has been *quarantined* has its outbound traffic limited, and all inbound traffic to it stopped.  This
```

Comment:
Communication with the Carbon Black Cloud backend is not limited or stopped otherwise we wouldn't be able to un-quarantine

Reply:
I took a lot of these descriptions from the user guide. The user guide didn't mention communication with CBC itself.